### ISSUE TITLE
Use ValueNotifier and remove provider 

### ISSUE THREAD

**Issue by deandreamatias** [YoussefRaafatNasry/lo_form#7]
To improve this package and remove all dependencies that can be, I suggest remove provider dependency and use a [ValueNotifier](https://api.flutter.dev/flutter/foundation/ValueNotifier-class.html) to update widgets states.
Maybe this [article](https://chowdera.com/2021/04/20210403224120163v.html) or this [other](https://medium.com/swlh/use-valuenotifier-like-pro-6441d9ddad05) can be help.

This this change will can improve:
- Avoid dependency conflicts when used this package
- No need update package every time than update provider ([like this](https://github.com/YoussefRaafatNasry/lo_form/issues/3))
- Decrease package size (even more!)
- Don't need use `notifyListeners()`

PS: I can work on this, but can't be promise when hehehe

**Comment by YoussefRaafatNasry** [comment]
I like the idea of dropping dependency on provider. But changing `ChangeNotifier` to `ValueNotifier` won't do it. The primary need for provider is to [access form's state inside the fields widgets](https://github.com/YoussefRaafatNasry/lo_form/blob/master/package/lib/src/core/lo_field.dart#L43-L48). We might need to use an `InheritedWidget` instead to achieve this.

**Comment by deandreamatias** [comment]
ValueListenableBuilder can be help with this read?

**Comment by YoussefRaafatNasry** [comment]
I don't think this is possible. `ValueListenableBuilder` is like the `Consumer` widget from provider package, which rebuilds it's children when the value changes only. But we cannot use it to access a value from ancestor widgets.

**Comment by YoussefRaafatNasry** [comment]
@deandreamatias I've managed to remove provider completely in 499d9d3e22ad64492350b33922c7aee6db99b74a.

**Comment by deandreamatias** [comment]
Great achievement!
